### PR TITLE
Flywheel/Shooter code and controls

### PR DESCRIPTION
**Pull request by BenWertz**
Code now works on the complete turret (minus the turret, that is):

The flywheel uses _both_ motors and has PID (still a WIP, it undershoots by a lot, but it at least resists the load of the power cell when shooting [for the most part {it's not perfect, but that's okay}]), and the hood is using the Hall Effect sensor but it has closed-loop position (no indexing _right_ now, but I'm pretty sure it could work if I uncommented it [didn't want to risk it]).

It also has a bunch of diagnostic displays in ShooterControls.java.

### CODE REVIEW

**Review by jjesse1972 — DISMISSED** [review]

**Review by ClaudP — CHANGES_REQUESTED** [review]
Please leave in Turret Indexing when merging into main unless absolutely necessary to the flywheel's current function.

**Review by ClaudP — APPROVED** [review]

**Review by classygirl44 — COMMENTED** [review]
Looks good!

Joke for the day: What do unicorns call their dads?

File: src/main/java/frc/robot/commands/ShooterControls.java [hunk 1]
```diff
@@ -26,21 +34,100 @@ public ShooterControls(ShooterInterface shooter_) {
...
+    // }
+    // if (OI.driverController.getBumperReleased(Hand.kLeft)) {
+    //   shooter.resetHood();
+    // }
+
+
+    // shooter.setHoodPower(0.1 * OI.driverController.getRawAxis(1));
+    
+    
+    shooter.setHoodAngle((shooter.getMaxHoodAngle() + shooter.getMinHoodAngle()) * 0.5
```
Comment (r-wallis) [inline src/main/java/frc/robot/commands/ShooterControls.java:86]: should there be a comment explaining this?

File: src/main/java/frc/robot/subsystems/instances/Shooter.java [hunk 2]
```diff
@@ -32,80 +35,130 @@
...
   private static double flywheelVelocity = 0;
   private static final double hoodIndexAngle = 0;
+  private static double hoodVelocityTarget = 0;
+  private static double hoodInputPower = 0;
 
   @Override
   public void periodic() {
-    isHoodIndexed = hoodIndexer.get();
+    if (OI.driverController.getStartButtonPressed()) {
+      isHoodIndexed = !isHoodIndexed;
```
Comment (r-wallis) [inline src/main/java/frc/robot/subsystems/instances/Shooter.java:138]: does this toggle or set to 0/false?

**Review by r-wallis — APPROVED** [review]

**Pull request merged**